php-checks/src/main/java/org/sonar/php/checks/NestedTernaryOperatorsCheck.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.expression.ConditionalExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionExpressionTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = "S3358")
+public class NestedTernaryOperatorsCheck extends PHPVisitorCheck {
+
+  @Override
+  public void visitConditionalExpression(ConditionalExpressionTree tree) {
+    TernaryVisitor visitor = new TernaryVisitor();
+    tree.condition().accept(visitor);
+    ExpressionTree trueExpression = tree.trueExpression();
+    if (trueExpression != null) {
+      trueExpression.accept(visitor);
+    }
+    tree.falseExpression().accept(visitor);
+    if (visitor.subTernary != null) {
+      context()
+        .newIssue(this, visitor.subTernary, "Extract this nested ternary operation into an independent statement.")
+        .secondary(tree.queryToken(), "Parent ternary operator");
+    }
+    // skip nested
+  }
+
+  private class TernaryVisitor extends PHPVisitorCheck {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)